lxd/network/network_utils.go

```diff
@@ -30,26 +30,26 @@ import (
 	"github.com/lxc/lxd/shared/logger"
 )
 
-// ValidNetworkName validates network name.
-func ValidNetworkName(value string) error {
+// validRealNetworkName validates a real network interface name.
+func validRealNetworkName(value string) error {
```

Comment:
Should we rename that to `validInterfaceName` instead to make it clear we care about validating what can be used as a network interface on Linux?

Reply:
I did consider that, the only thing that stopped me was the check for veth prefix. Im not sure why that was there, but seemed specific to networks rather than all nic interfaces

Reply:
Yeah, not sure why we're checking for the veth prefix, it's obviously not a good idea to use it but it's not different than use ethX or tapXXX, I suspect we can just drop that part of the check.

Reply:
@stgraber ok thats fixed now.